sigstore/_internal/tuf.py

```diff
@@ -19,12 +19,14 @@
 from __future__ import annotations
 
 import logging
+from functools import lru_cache
 from pathlib import Path
 from urllib import parse
 
 import appdirs
 from cryptography.x509 import Certificate, load_pem_x509_certificate
 from tuf.ngclient import Updater
+from tuf.ngclient._internal.requests_fetcher import RequestsFetcher
```

Comment:
NB: This makes use of an internal TUF API, although I think this is acceptable for now: this internal API shows up in public interfaces and is unlikely to change in the near future; we also may be able to get `python-tuf` to make it into a public API.

Reply:
xref https://github.com/theupdateframework/python-tuf/issues/2268